apps/frontend/src/app/Database/DataManagers/ArtifactData.ts

```diff
@@ -128,15 +112,22 @@ export class ArtifactDataManager extends DataManager<string, "artifacts", ICache
           }
           isUpgrade ? result.artifacts.upgraded.push(art) : result.artifacts.unchanged.push(art)
           idsToRemove.delete(match.id)
-          if (importId) {
-            delete this.data[match.id]
-            this.removeStorageEntry(match.id)
-          } else importId = match.id
+          if (importId) this.remove(match.id, false)// Do not notify, since this is a "replacement"
+          else importId = match.id
           importArt = { ...art, location: hasEquipment ? art.location : match.location }
         }
       }
       if (importId) {
-        swapId(importId, generateRandomArtID(takenIds))
+        if (this.get(importId)) { // swap existing to another id
+          const newId = this.generateKey(takenIds)
+          takenIds.add(newId)
+          this.swapId(importId, newId)
+
+          if (idsToRemove.has(importId)) {
```

Comment:
What is this for?

Reply:
`idsToRemove` is basically a list of ids that does not exist in the new import. 
This logic keeps this list synced, when a swap(line 124) happens.

Reply:
```suggestion
          // We planned to remove this id earlier. Need to update the ID to the newly changed one
          if (idsToRemove.has(importId)) {
```